src/sympc/protocol/spdz/spdz.py

```diff
@@ -52,30 +53,23 @@ def mul_master(
         raise ValueError(f"{op_str} should be in {EXPECTED_OPS}")
 
     session = x.session
-
     shape_x = tuple(x.shape)
     shape_y = tuple(y.shape)
 
-    args = [
-        list(el) + [op_str]
-        for el in zip(session.session_ptrs, x.share_ptrs, y.share_ptrs)
-    ]
-
-    try:
-        mask = parallel_execution(spdz_mask, session.parties)(args)
-    except EmptyPrimitiveStore:
-        CryptoPrimitiveProvider.generate_primitives(
-            f"beaver_{op_str}",
-            sessions=session.session_ptrs,
-            g_kwargs={
-                "a_shape": shape_x,
-                "b_shape": shape_y,
-                "nr_parties": session.nr_parties,
-                **kwargs_,
-            },
-            p_kwargs={"a_shape": shape_x, "b_shape": shape_y},
-        )
-        mask = parallel_execution(spdz_mask, session.parties)(args)
+    args = [list(el) + [op_str] for el in zip(x.share_ptrs, y.share_ptrs)]
+
+    CryptoPrimitiveProvider.generate_primitives(
+        f"beaver_{op_str}",
+        session=session,
+        g_kwargs={
+            "a_shape": shape_x,
+            "b_shape": shape_y,
+            "nr_parties": session.nr_parties,
+            **kwargs_,
+        },
+        p_kwargs={"a_shape": shape_x, "b_shape": shape_y},
+    )
+    mask = parallel_execution(spdz_mask, session.parties)(args)
 
     eps_shares, delta_shares = zip(*mask)
```

Comment:
We had this because we were not checking for pre-generated primitive and using the exception as a way to generated the primitive on the fly... will it still work ?

Reply:
Ahh...you are right, I removed this by mistake (the `try...catch` block)